File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcGroup/IfcSystem/IfcBuildingSystem/IfcBuildingSystem.vb
Public Class IfcBuildingSystem Inherits IfcSystem
    Public PredefinedType As IfcBuildingSystemTypeEnum
    Public LongName As IfcLabel

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcBuildingSystemTypeEnum.USERDEFINED) OR ((PredefinedType = IfcBuildingSystemTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
End Class

File: output/ifc4x3_dev_73740fe4/entity/IfcBuildingSystem.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


public class IfcBuildingSystem extends IfcSystem {

    public IfcBuildingSystemTypeEnum PredefinedType; // OPTIONAL
    public IfcLabel LongName; // OPTIONAL
    // INVERSE attributes:
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR
    // (PredefinedType <> IfcBuildingSystemTypeEnum.USERDEFINED) OR
    // ((PredefinedType = IfcBuildingSystemTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType));

    // WHERE constraints:
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR
    // (PredefinedType <> IfcBuildingSystemTypeEnum.USERDEFINED) OR
    // ((PredefinedType = IfcBuildingSystemTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType));
}
